newt/repo/repo.go

```diff
@@ -596,18 +499,18 @@ func (r *Repo) Sync(vers *Version, force bool) (bool, error) {
 		msg := "Unexpected local branch for %s: \"%s\" != \"%s\""
 		if force {
 			util.StatusMessage(util.VERBOSITY_DEFAULT,
-				msg+"\n", r.rdesc.name, currBranch, branchName)
+				msg+"\n", r.Name(), currBranch, branchName)
 		} else {
-			err = util.FmtNewtError(
-				msg, r.rdesc.name, currBranch, branchName)
+			err := util.FmtNewtError(
+				msg, r.Name(), currBranch, branchName)
```

Comment:
hmmmm, I thought the rule for short variable declaration was that at least one variable must be newly declared... but `err` is already declared above...

Reply:
Since this `err` is declared in a child scope, it shadows the earlier declaration.  This is analogous to the C code:

```
int err = 10;
while (keep_going()) {
    int err = 20;
    /* err == 20 */
}
/* err == 10 */
```

I think shadowing a variable declaration can be confusing, and is probably best avoided.  In the case of an `err` variable, I don't think there is that much room for confusion, however.  Though, in this case, I don't think the second `err` variable should be declared at all, so I am going to change it to:

```
            return false, util.FmtNewtError(
                msg, r.Name(), currBranch, branchName)
```